src/FINDOLOGIC/Api/Responses/Response.php

```diff
@@ -125,10 +124,24 @@ private static function buildSearchOrNavigationResponse(
     /**
      * Checks if the response is valid. If not, an exception will be thrown.
      *
+     * @param Request $request
      * @param GuzzleResponse $response
      */
-    protected static function checkResponseIsValid($response)
+    protected static function checkResponseIsValid(Request $request, GuzzleResponse $response)
     {
+        if ($request->getOutputAdapter() === OutputAdapter::XML_21) {
+            $xml= new DOMDocument();
+            $xml->loadXML($response->getBody()->getContents());
+
+            libxml_use_internal_errors(true);
```

Comment:
Move method `loadXML` after `libxml_use_internal_errors(true)`, because even this method can throw warnings. Also avoid confusing newlines when structurally they should be together.